src/main/scala/org/broadinstitute/dsde/firecloud/model/ElasticSearch.scala

```diff
@@ -37,38 +45,76 @@ object Document {
 
 // classes to convert from json body and to json response
 case class LibrarySearchParams(
-  searchTerm: Option[String],
+  searchString: Option[String],
+  searchFields: Map[String, Seq[String]],
+  fieldAggregations: Seq[String],
+  maxAggregations: Option[Int],
   from: Int = 0,
   size: Int = 10)
 
 object LibrarySearchParams {
-  def apply(searchTerm: Option[String], from: Option[Int], size: Option[Int]) = {
-    new LibrarySearchParams(searchTerm, from.getOrElse(0), size.getOrElse(10))
+  def apply(searchTerm: Option[String], fieldTerms: Map[String, Seq[String]], fieldAggregations: Seq[String], maxAggregations: Option[Int], from: Option[Int], size: Option[Int]) = {
+    new LibrarySearchParams(searchTerm, fieldTerms, fieldAggregations, maxAggregations, from.getOrElse(0), size.getOrElse(10))
   }
 }
 
 case class LibrarySearchResponse(
   searchParams: LibrarySearchParams,
   total: Int,
-  results: Seq[JsValue]) {
-}
+  results: Seq[JsValue],
+  aggregations: Seq[LibraryAggregationResponse])
+
+case class LibraryAggregationResponse(
+  field: String,
+  results: AggregationFieldResults)
+
+case class AggregationFieldResults(  numOtherDocs: Int,
+  buckets: Seq[AggregationTermResult])
 
+case class AggregationTermResult(key: String, doc_count: Int)
 
 // classes to create the ES queries in json format
 // {"query":{"match_all":{}}}"
-// {"query":{"wildcard":{"_all":"%s"}}}"
+
+//{"query":{
+//  "constant_score":{
+//  "filter":{
+//  "bool":{
+//  "must":[
+//{"bool":{
+//  "should":[
+//{"term":{"library:indication":"n/a"}},
+//{"term":{"library:indication":"disease"}},
+//{"term":{"library:indication":"lukemia"}}]
+//}
+//},
+//{"match":{"all_":"broad"}}]}}}}}
+
+
 trait QueryMap
+trait ESLogicType
 
-case class ESQuery(query: QueryMap)
+case class ESQuery(query: ESConstantScore)
 
-case class ESMatchAll(match_all: Map[String,String]) extends QueryMap {
-  def this() = this(Map.empty[String,String])
-}
+case class ESConstantScore(constant_score: ESFilter)
+
+case class ESFilter(filter: ESBool)
+
+case class ESBool(bool: ESLogicType) extends QueryMap
+
+case class ESMust(must: Seq[QueryMap]) extends ESLogicType
+
+case class ESShould(should: Seq[QueryMap]) extends ESLogicType
 
-case class ESWildcard(wildcard: ESWildcardSearchTerm) extends QueryMap {
-  def this(term: String) = this(ESWildcardSearchTerm(term))
+case class ESMatch(`match`: Map[String, String]) extends QueryMap {
+  // when the search term should search all columns/attributes
+  def this(value: String) = this(Map("_all" -> value))
 }
 
-case class ESWildcardSearchTerm(_all: String)
+case class ESTerm(term: Map[String, String]) extends QueryMap
+
+case class ESMatchAll(match_all: Map[String,String]) extends QueryMap {
+  def this() = this(Map.empty[String,String])
+}
```

Comment:
no problem - would it be useful to add some debug level logging statements to print this out?